monitor/plugins/vertical/plugin.py

```diff
@@ -0,0 +1,156 @@
+
+        application_progress_error = {}
+        cpu_usage_metric = {}
+        cpu_quota_metric = {}
+
+        # Reference Value
+        ref_value = float(self.cpu_threshold)
+        
+        # Error
+        
+        print("CPU_USAGE: " + str(cpu_usage))
+        print("REF_VALUE: " + str(ref_value))
+
+        error = (float(cpu_usage)/100) - ref_value
+
+        application_progress_error['name'] = ('application-progress'
+                                              '.error')
+        application_progress_error['value'] = error
+        application_progress_error['timestamp'] = time.time() * 1000
+        application_progress_error['dimensions'] = self.dimensions
+
+        cpu_usage_metric['name'] = 'cpu-usage'
+        cpu_usage_metric['value'] = float(cpu_usage)
+        cpu_usage_metric['timestamp'] = time.time() * 1000
+        cpu_usage_metric['dimensions'] = self.dimensions
+
+        cpu_quota = self.get_cpu_quota()
+
+        cpu_quota_metric['name'] = 'cpu-quota'
+        cpu_quota_metric['value'] = float(cpu_quota)
+        cpu_quota_metric['timestamp'] = time.time() * 1000
+        cpu_quota_metric['dimensions'] = self.dimensions
+
+        print "Error: %s " % application_progress_error['value']
+
+        self.rds.rpush(self.metric_queue,
+                       str(application_progress_error))
+
+        if self.enable_visualizer:
+            self.datasource.send_metrics([application_progress_error])
+            self.datasource.send_metrics([cpu_usage_metric])
+            self.datasource.send_metrics([cpu_quota_metric])
+            
+        time.sleep(MONITORING_INTERVAL)
+
+    def _get_elapsed_time(self):
+        datetime_now = datetime.now()
+        elapsed_time = datetime_now - self.submission_time
+        print "Elapsed Time: %.2f" % elapsed_time.seconds
+
+        return elapsed_time.seconds
+
+    def monitoring_application(self):
+        try:
+            cpu_usage = requests.get('http://%s:5000' % (self.get_api_address())).text
+            
+            print("Publishing metric %s value %s: " % (self.metric_source, cpu_usage))
+
+            self._publish_measurement(cpu_usage=cpu_usage)
+
+        except Exception as ex:
+            print ("Error: No application found for %s. %s remaining attempts"
+                   % (self.app_id, self.attempts))
+
+            print ex.message
+            raise
+
+    def get_cpu_quota(self):
+        try:
+            cpu_quota = requests.get('http://%s:5000/%s' % (self.get_api_address(), self.get_metric_endpoint)).text
+            return cpu_quota
+
+        except Exception as ex:
+            print("Error while getting %s metric" % (self.metric_source))
+            print ex.message
+            raise
+
+    def get_api_address(self):
+        # TODO Search a better way to get the internal ip of a node using k8s api
+        nodes_ips = os.popen("kubectl --kubeconfig=%s get nodes -o wide | awk '{print $6}'" % (self.k8s_manifest)).readlines()
```

Comment:
use python kubernetes client instead. we should not just assume the user has `kubectl` installed in monitor's host.